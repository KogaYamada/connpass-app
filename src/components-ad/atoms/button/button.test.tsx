import React from 'react';
import Button from './button.component';
import { shallow,configure, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('button snapshot tests', () => {
  let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;
  beforeEach(() => {
    wrapper=shallow(<Button title={'ボタン'} />);
  })

  it('<Button title={"ボタン"}/>: title以外のpropsなし', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('<Button title={"ボタン"} size="large"/>: サイズlarge', () => {
    wrapper.setProps({size: 'large'})
    expect(wrapper).toMatchSnapshot();
  });

  it('<Button title={"ボタン"} size="middle"/>: サイズmiddle', () => {
    wrapper.setProps({size: 'middle'})
    expect(wrapper).toMatchSnapshot();
  });

  it('<Button title={"ボタン"} size="small"/>: サイズsmall', () => {
    wrapper.setProps({size: 'small'})
    expect(wrapper).toMatchSnapshot();
  });
});
