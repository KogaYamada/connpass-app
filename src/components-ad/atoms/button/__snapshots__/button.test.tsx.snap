// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button snapshot tests <Button title={"ボタン"} size="large"/>: サイズlarge 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 10,
        "borderWidth": 1,
        "paddingHorizontal": 10,
      },
      Object {
        "minWidth": 250,
        "paddingVertical": 17,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    ボタン
  </Text>
</ForwardRef>
`;

exports[`button snapshot tests <Button title={"ボタン"} size="middle"/>: サイズmiddle 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 10,
        "borderWidth": 1,
        "paddingHorizontal": 10,
      },
      Object {
        "maxWidth": 250,
        "minWidth": 150,
        "paddingVertical": 13,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    ボタン
  </Text>
</ForwardRef>
`;

exports[`button snapshot tests <Button title={"ボタン"} size="small"/>: サイズsmall 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 10,
        "borderWidth": 1,
        "paddingHorizontal": 10,
      },
      Object {
        "maxWidth": 150,
        "paddingVertical": 10,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    ボタン
  </Text>
</ForwardRef>
`;

exports[`button snapshot tests <Button title={"ボタン"}/>: title以外のpropsなし 1`] = `
<ForwardRef
  style={
    Array [
      Object {
        "backgroundColor": "#F2F2F2",
        "borderColor": "#DBDBDB",
        "borderRadius": 10,
        "borderWidth": 1,
        "paddingHorizontal": 10,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontSize": 18,
          "textAlign": "center",
        },
      ]
    }
  >
    ボタン
  </Text>
</ForwardRef>
`;
